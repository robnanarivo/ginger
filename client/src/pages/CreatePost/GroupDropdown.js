import { React, useState } from 'react';
import {
  InputLabel,
  MenuItem,
  FormHelperText,
  FormControl,
  Select,
} from '@mui/material';

const GroupDropdown = ({ groupList, defaultGroup }) => {
  const [dropdown, setDropdown] = useState(defaultGroup);

  // post into a chosen group
  const handleChange = (e) => {
    setDropdown(e.target.value);
  };

  return (
    <FormControl sx={{ width: 400, mt: 2, mb: 1 }} required>
      <InputLabel>Group</InputLabel>
      <Select label="Group" name="group" value={dropdown} onChange={handleChange}>
        {groupList.map((group) => (
          <MenuItem key={group.id} value={group.id}>{group.name}</MenuItem>
        ))}
      </Select>
      <FormHelperText>Select a group to post into</FormHelperText>
    </FormControl>
  );
};

export default GroupDropdown;
