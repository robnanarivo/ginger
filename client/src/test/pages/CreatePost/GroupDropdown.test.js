import React from 'react';
import renderer from 'react-test-renderer';
import GroupDropdown from '../../../pages/CreatePost/GroupDropdown';
import 'jest-styled-components';

describe('GroupDropdown test', () => {
  it('GroupDropdown Snapshot test', () => {
    const groupList = [
      { id: 123, name: 'cool stuff' },
    ];
    const tree = renderer
      .create(<GroupDropdown groupList={groupList} defaultGroup={123} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
