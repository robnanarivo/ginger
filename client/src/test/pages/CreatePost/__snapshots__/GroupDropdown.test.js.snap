// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupDropdown test GroupDropdown Snapshot test 1`] = `
<div
  className="MuiFormControl-root css-1kvrxa3-MuiFormControl-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
  >
    Group
    <span
      aria-hidden={true}
      className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
    >
       
      *
    </span>
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="mui-component-select-group"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mui-component-select-group"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      cool stuff
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="group"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={true}
      tabIndex={-1}
      value={123}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1z7n62"
      >
        <span>
          Group
           
          *
        </span>
      </legend>
    </fieldset>
  </div>
  <p
    className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled Mui-required css-1wc848c-MuiFormHelperText-root"
  >
    Select a group to post into
  </p>
</div>
`;
